### ISSUE TITLE
homegear doesn't compile

### ISSUE THREAD

**Issue by jcw9930** [Homegear/Homegear#238]
Currently neither 0.5.25 nor git master compile for me:

0.5.25:
```
$ make config=release -j2
==== Building base (release) ====
==== Building homematicbidcos (release) ====
HelperFunctions.cpp
Linking homematicbidcos
/usr/bin/ld: cannot find -lbase
collect2: error: ld returned 1 exit status
homematicbidcos.make:140: recipe for target 'lib/Modules/Release/libhomematicbidcos.so' failed
make[1]: *** [lib/Modules/Release/libhomematicbidcos.so] Error 1
Makefile:21: recipe for target 'homematicbidcos' failed
make: *** [homematicbidcos] Error 2
make: *** Waiting for unfinished jobs....
Variable.cpp
Modules/Base/HelperFunctions/HelperFunctions.cpp: In static member function 'static bool BaseLib::HelperFunctions::fileExists(std::string)':
Modules/Base/HelperFunctions/HelperFunctions.cpp:96:9: error: cannot convert 'std::ifstream {aka std::basic_ifstream<char>}' to 'bool' in return
  return in;
         ^
base.make:212: recipe for target 'obj/Release/base/HelperFunctions.o' failed
make[1]: *** [obj/Release/base/HelperFunctions.o] Error 1
make[1]: *** Waiting for unfinished jobs....
Makefile:17: recipe for target 'base' failed
make: *** [base] Error 2
```

git master:
```
g++ -DHAVE_CONFIG_H -I. -I..  -Wall -std=c++11 -DFORTIFY_SOURCE=2 -DGCRYPT_NO_DEPRECATED  -I/usr/include/php7-homegear -I/usr/include/php7-homegear/main -I/usr/include/php7-homegear/sapi -I/usr/include/php7-homegear/TSRM -I/usr/include/php7-homegear/Zend -DLINUXSYSTEM  -g -O2 -MT ScriptEngine/libscriptengine_a-php_sapi.o -MD -MP -MF ScriptEngine/.deps/libscriptengine_a-php_sapi.Tpo -c -o ScriptEngine/libscriptengine_a-php_sapi.o `test -f 'ScriptEngine/php_sapi.cpp' || echo './'`ScriptEngine/php_sapi.cpp
In file included from ScriptEngine/../GD/GD.h:36:0,
                 from ScriptEngine/php_sapi.cpp:31:
ScriptEngine/../GD/../ScriptEngine/ScriptEngine.h:35:35: fatal error: homegear-base/BaseLib.h: No such file or directory
compilation terminated.
```

This is on Arch Linux

**Comment by hfedcba** [comment]
I will look into the issue with version 0.5.25. For the current master branch you need to compile the base library first (libhomegear-base). You also need PHP 7 compiled as a static labrary. The compilation process is a little tricky, A tutorial is coming ;-).

Cheers,

Sathya

**Comment by mapero** [comment]
I tried to compile this on archlinux. I was able to create a package for libhomegear-base. But i fail to build this repository, mainly because of the php7 dependency. Using php7-embed from aur and patching the makefiles to replace "php7-homegear" with php7 fails. Is there something special about your version ("php7-homegear")? Where is the project/source of this version? Any hints?

Thanks.

**Comment by hfedcba** [comment]
Hey,

does the information here help?

https://homegear.eu/doc/en-US/html/sect-Homegear-Homegear_0.6_Documentation-Compiling-from-Source_Manual.html

Cheers,

Sathya

**Comment by jcw9930** [comment]
is it really necessary to depend on a patched version of php? it makes deployment on small/embedded hw more difficult

**Comment by hfedcba** [comment]
No. I just checked, the patch of the configure file isn't necessary anymore. But currently pthreads still needs to be patched. pthreads is needed for device definitions over XML to be able to handle events without polling (see for example https://github.com/Homegear/Homegear-OpenWeatherMap/blob/master/OpenWeatherMap.xml). But I now created an issue to see if we can achieve this without patching: https://github.com/krakjoe/pthreads/issues/557.
